projects/indicators/src/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -112,50 +127,45 @@ export class SkyTextHighlightDirective
     return regex.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
   }
 
-  private static getRegexMatch(
-    node: HTMLElement,
-    searchText: string
-  ): RegExpExecArray {
-    const text = node.nodeValue;
-    const newSearchText = this.cleanRegex(searchText);
-    const searchRegex = new RegExp(newSearchText, 'gi');
-
-    return searchRegex.exec(text);
-  }
-
-  private static markNode(node: any, searchText: string): number {
-    const regexMatch = SkyTextHighlightDirective.getRegexMatch(
-      node,
-      searchText
-    );
+  private static markNode(node: any, searchTerms: string[]): number {
+    /* istanbul ignore else */
+    if (searchTerms) {
+      const text = node.nodeValue;
+      for (let index = 0; index < searchTerms.length; index++) {
+        searchTerms[index] = this.cleanRegex(searchTerms[index]);
+      }
+      const searchRegex = new RegExp(searchTerms.join('|'), 'gi');
 
-    // found match
-    if (regexMatch && regexMatch.length > 0) {
-      // split apart text node with mark tags in the middle on the search term
-      const matchIndex = regexMatch.index;
+      let match;
+      while ((match = searchRegex.exec(text))) {
```

Comment:
Now that we're looping and replacing, I'm wondering what would happen if you had two terms where the first was a substring of the second (e.g. `['abc', 'abcd']`). If `abc` is replaced, will `abcd` be found and replaced as well? Or will it not be found because `abc` has been replaced with `<mark>abc</mark>`?

Reply:
Good thought. It will NOT find a match if an entry is a substring of another entry appearing later in the array. This is actually how regex works too, strangely.... `/pan|panda/ig` will only match on `pan` and completely miss any `panda` matches.
I started looking into filtering out any substrings that appear in the array, but it took a couple of loops and started to feel like a lot of heavy logic for small payoff. In looking for a more lightweight solution, I found that you can also sort the array in descending order, which is takes much less logic and achieves the same result. I've added that logic to the setter. What do you think?

Reply:
I think this solution will work, but I'd recommend updating the comment to say that you're ordering the strings by their _length_ in descending order, not the string values themselves, as that tripped me up during the review.

Reply:
Done.

Reply:
The `while` loop with a `return` statement is not going to loop more than once. The return needs to be moved outside the loop and use another variable to track whether there was a match.